executor/select_into.go

```diff
@@ -52,7 +52,7 @@ func (s *SelectIntoExec) Open(ctx context.Context) error {
 		return errors.New("unsupported SelectInto type")
 	}
 
-	f, err := os.OpenFile(s.intoOpt.FileName, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
+	f, err := os.OpenFile(s.intoOpt.FileName, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0600)
```

Comment:
When running `SELECT 1 INTO OUTFILE '/tmp/foo.out'` this results in:
```
$ ls -lh /tmp/foo.out;
-rw-rw-r--. 1 dvaneeden dvaneeden 2 Jan 19 07:40 /tmp/foo.out
```

Note that `tidb-server` in this case is running as the `dvaneeden` user, but in production this would probably run as the `tidb` user. Having wider permissions here allows the user executing the SQL to then read the file. With this change one would probably need sudo to be able to do this.

With MySQL 8.0.27 the file permissions are like this:
```
$ ls -lh /tmp/bar.out
-rw-r-----. 1 dvaneeden dvaneeden 2 Jan 19 07:44 /tmp/bar.out
```
(also with `mysqld` running as `dvaneeden`).

So probably good to do what MySQL does and allow the file to be group readable. This would allow someone to be added to the `mysql` or `tidb` group and still read the file.

```suggestion
	f, err := os.OpenFile(s.intoOpt.FileName, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0640)
```

If we want to make this more secure we should implement [`secure_file_priv`](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_secure_file_priv) ( #11582 )